lib/background_streamer/worker.rb

```diff
@@ -3,142 +3,111 @@
 
 module BackgroundStreamer
   class Worker
-    CRLF= "\r\n".freeze
+    CRLF = "\r\n".freeze
+    RACK_HIJACK = 'rack.hijack'.freeze
+    RACK_HIJACK_IO = 'rack.hijack_io'.freeze
+    ACTION_DISPATCH_REQUEST_ID = 'action_dispatch.request_id'.freeze
+    X_REQUEST_ID = 'X_REQUEST_ID'.freeze
 
-    @count = 0
-    @middleware = []
+    attr_reader :options, :timeout, :request_id, :status, :headers, :io, :body
 
     class << self
-      attr_accessor :count
-      attr_accessor :middleware
-      def inc
-        @count += 1
-      end
+      def perform_async(env, body, options={})
+        env[RACK_HIJACK].call
+
+        Thread.new do 
```

Comment:
Good point.  The previous implementation was using a sized queue (default of 50) to prevent a certain amount of work from being queued up.  Perhaps we should track the threads we create to determine if they're still processing and are not above some definable threshold.